cmake/external/mkldnn.cmake

```diff
@@ -92,15 +84,15 @@ add_definitions(-DPADDLE_WITH_MKLDNN)
 SET(dummyfile ${CMAKE_CURRENT_BINARY_DIR}/mkldnn_dummy.c)
 FILE(WRITE ${dummyfile} "const char * dummy = \"${dummyfile}\";")
 ADD_LIBRARY(mkldnn STATIC ${dummyfile})
-TARGET_LINK_LIBRARIES(mkldnn ${MKLDNN_LIB} ${MKLML_LIB} ${MKLML_IOMP_LIB})
+TARGET_LINK_LIBRARIES(mkldnn ${MKLDNN_LIB} ${MKLML_IOMP_LIB})
 ADD_DEPENDENCIES(mkldnn ${MKLDNN_PROJECT})
 
 # copy the real so.0 lib to install dir
 # it can be directly contained in wheel or capi
 if(WIN32)
     SET(MKLDNN_SHARED_LIB ${MKLDNN_INSTALL_DIR}/bin/mkldnn.dll)
 else(WIN32)
-    SET(MKLDNN_SHARED_LIB ${MKLDNN_INSTALL_DIR}/libmkldnn.so.0)
+    SET(MKLDNN_SHARED_LIB ${MKLDNN_INSTALL_DIR}/libmkldnn.so.1)
```

Comment:
How about
- copy `libmkldnn.so.1` to `libmkldnn.so.0` in `MKLDNN_INSTALL_DIR` first.
- then, `SET(MKLDNN_SHARED_LIB ${MKLDNN_INSTALL_DIR}/libmkldnn.so.0)`

Reply:
It's not so easy because we don't decide which .so to choose, its MKL-DNN's core job. If we link in paddle libmkldnn.so.0 (event with new "code"), any MKL-DNN operator will crash with error code "ImportError: libmkldnn.so.1: cannot open shared object file: No such file or directory"

Reply:
I guess the luotao's problem is, when paddle is updated with new mkldnn version, users may need to re-compile their apps, it is expected to be avoided.

@luotao1 in practice, if paddle is updated, need users do re-compile?

Reply:
users will re-compile, but they don't want to change the CMakeLists.txt. That is to say, we want the structure of `fluid_inference.tgz` is stable.

Maybe you can change the name in the `inference_lib.cmake`
https://github.com/PaddlePaddle/Paddle/blob/c0dcb090a3fa239ccfa5ea8821dd05c98829bcc2/cmake/inference_lib.cmake#L98-L109